### DISCUSSION TITLE
Displaying Test Data on a Label

### DISCUSSION THREAD

**Discussion by vk2tds** [inventree/InvenTree#7496] [General]
I am attempting to display test data on a label for a serialised stock item. The stock item label template appears below. 

The resulting PDF has the QR code, serial number and name as it should. Alas, it cannot find the test data. I have tried a number of times in attempt to work out where I was going wrong, and I am stuck. I copied the `for` statement from the test report, modifying it to the most basic form, also without success. 

I just cannot work out what I am doing wrong. Any thoughts would be appreciated. Thanks

```
{% extends "label/label_base.html" %}

{% load l10n i18n barcode %}

{% block style %}

.qr {
    position: absolute;
    left: 0mm;
    top: 0mm;
    {% localize off %}
    height: {{ height }}mm;
    width: {{ height }}mm;
    {% endlocalize %}
}

.part {
    font-family: Arial, Helvetica, sans-serif;
    display: flex;
    position: absolute;
    {% localize off %}
    left: {{ height }}mm;
    {% endlocalize %}
    top: 2mm;
}

{% endblock style %}

{% block content %}

<img class='qr' alt="{% trans 'QR Code' %}" src='{% qrcode qr_data %}'>


<div class='part'>
	S/N: {{ serial }}
	{% if test_keys|length > 0 %}	
		tests found
	{% else %}
		tests not found
	{% endif %}
    {% for key in test_keys %}
		{{ key }}
    {% endfor %}
	{{ name }}
    <!-- {{ part.full_name }} -->

</div>

{% endblock content %}
``` 

![Screenshot 2024-06-23 at 6 35 53 pm](https://github.com/inventree/InvenTree/assets/682818/2b6a86e4-cbea-44c2-ab7a-accdcdde2482)

![Screenshot 2024-06-23 at 6 37 13 pm](https://github.com/inventree/InvenTree/assets/682818/ad02d803-c553-4f1a-9ed2-d88b92008705)

**Comment by SchrodingersGat** [comment]
Your approach seems sound, looking at the available "context" variables for the stock item report: https://docs.inventree.org/en/latest/report/context_variables/#stock-item

What version of InvenTree server are you running?

**Reply by vk2tds** [reply]
I am using 0.15.3
